ccube.nim

```diff
@@ -66,142 +66,101 @@ Examples:
   file name pp.txt.
 
   cc* = collatz conj
-""")
+"""
   quit()
 
-proc parseParams():(string,string,string) =
-  if paramCount() > 0:
-    var
-      paramSequence:seq[string] = @[]
-      forbiddenCharArray = ["A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","Z","!","@","#","$","%","^","&","*","(",")","_","-","=","+","'","\"",":",";","<",">",".","/","?","[","]","{","}"]
-      validBools = ["true","false"]
-      noLoopSwitch = "--noloop:"
-      fileSwitch = "--file:"
-      numberParam = "-n:"
-      filePath:string
-      number:string
-      noLoopSwitchBool:string
-    for i in 1..paramCount():
-      paramSequence.add(paramStr(i))
-    for param in paramSequence:
-      if noLoopSwitch in param:
-        try:
-          noLoopSwitchBool = param.split(':')[1]
-          if noLoopSwitchBool in validBools:
-            discard
-          else:
-            stderr.write(&"Error:\"{noLoopSwitchBool}\",Invalid bool used")
-            quit()
-        except IndexDefect:
-          stderr.write("Error:Bad use of --noloop switch,valid bool expected")
-          quit()
-      
-      if numberParam in param:
-        try:
-          number = param.split(':')[1]
-          for forbiddenChar in forbiddenCharArray:
-            if forbiddenChar.toUpper in number or forbiddenChar.toLower in number:
-              stderr.write(fmt"Error:Expected a valid integer but got '{number}',[Value Error]")
-              quit()
-        except IndexDefect:
-          stderr.write("Error:Bad use of --noloop switch,valid integer expected")
-          quit()
-      
-      if fileSwitch in param:
+
+type
```

Comment:
This is a type that holds all program options which is clearer than passing tuples, and you'll have to change less code if you need to change the options